python/GafferUITest/ColorChooserTest.py

```diff
@@ -0,0 +1,92 @@
+##########################################################################
+#
+#  Copyright (c) 2024, Cinesite VFX Ltd. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import pathlib
+import unittest
+
+import imath
+
+import IECoreImage
+
+import Gaffer
+import GafferUI
+from GafferUI.ColorChooser import _tmi2rgb
+from GafferUI.ColorChooser import _rgb2tmi
+import GafferUITest
+
+class ColorChooserTest( GafferUITest.TestCase ) :
+
+	def testTMI( self ) :
+		# Load a precomputed image generated by Houdini to test our TMI <-> RGB calculation.
+		# The image is 200px x 20px. Each 20px square varies the temperature from -1.0 to 1.0
+		# on the X axis and magenta from -1.0 to 1.0 on the Y-axis. Each of the 10 squares, from
+		# left to right, varies the intensity from 0.0 to 0.9.
+		tileWidth = 20
+		tileCount = 10
+		tmiTarget = IECoreImage.ImageReader( ( pathlib.Path( __file__ ).parent / "images" / "tmi.exr" ).as_posix() ).read()
```

Comment:
I'd like to avoid any further dependencies on ImageReader - we're getting close to being able to ditch it completely. Can we use OpenImageIO instead?

Reply:
Alternately, we could perhaps test without an image - my vague recollection was that that was the plan anyway.

Reply:
Unfortunately OpenImageIO tries to return image pixels as a numpy array and fails if it can't do so. I don't see any way to ask it for a regular array or access pixels one at a time.

I could do a brute-force script to serialize the colors from the EXR directly into Python to include in the test script. Or maybe write out a file just made up of the floats and read it as standard binary data?

Reply:
Is [getchannel](https://openimageio.readthedocs.io/en/latest/imagebuf.html#_CPPv4NK4OIIO8ImageBuf10getchannelEiiii8WrapMode) available in Python?

Reply:
Ah yes, that does the job. I'm using `getpixel()` in https://github.com/GafferHQ/gaffer/pull/5962/commits/846d22fe6cda3bf21a98cc6a50bc299f952ab53d because `getchannel()` was complaining about mismatched arguments even when I used the example from their documentation.